File: IAMenuController/IAScreenEdgeGestureRecognizer.m
/* IAScreenEdgeGestureRecognizer.m
 * IAMenuController
 *
 * Created by Gordon Fontenot on 10/22/13.
 * Copyright (c) 2013 Mark Adams. All rights reserved.
 */

#import <UIKit/UIGestureRecognizerSubclass.h>
#import "IAScreenEdgeGestureRecognizer.h"

@implementation IAScreenEdgeGestureRecognizer

- (id)initWithTarget:(id)target action:(SEL)action {
  self = [super initWithTarget:target action:action];
  if (!self) return nil;

  self.cancelsTouchesInView = YES;

  return self;
}

- (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event {
  UITouch *touch = [touches allObjects][0];
  CGPoint location = [touch locationInView:self.view];
  if (location.x > 50) {
    self.state = UIGestureRecognizerStateFailed;
    return;
  }

  [super touchesBegan:touches withEvent:event];
}

@end

File: IAMenuController/IAScreenEdgeGestureRecognizer.swift
/*  IAScreenEdgeGestureRecognizer.swift
 *  IAScreenEdgeGestureRecognizer
 *
 *  Created by Taylor Wright-Sanson on 9/7/18.
 *  Created by Mark Adams on 12/16/11.
 *  MIT License
 */

import UIKit.UIGestureRecognizerSubclass

@objc public class IAScreenEdgeGestureRecognizer: UIPanGestureRecognizer {

  override init(target: Any?, action: Selector?) {
    super.init(target: target, action: action)

    cancelsTouchesInView = true
  }

  override public func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent) {
    if let touch = touches.first {
      let location = touch.location(in: view)
      if location.x > 50 {
        state = .failed
        return
      }
    }

    super.touchesBegan(touches, with: event)
  }
}
